app/models/user.rb

```diff
@@ -1,3 +1,29 @@
 class User < ApplicationRecord
   belongs_to :project
+  scope :users_joins_project, -> {includes(:project)}
+
+  def self.filter(id)
+    @users = id.to_i == 0 ? User.users_joins_project : Project.find(id.to_i).users.users_joins_project
+  end
+
+  def self.as_json(users)
```

Comment:
This methods destroys single responsibly. Create another class and put that method there.